# nf-core/fastqrepair: Output

## Introduction

This document describes the output produced by the pipeline.

## Pipeline overview

The nf-core/fastqrepair pipeline is built using [Nextflow](https://www.nextflow.io/). Output files are gathered into the `repaired` subfolder of the output folder set up using the `--outdir` command line parameter.

```
<output folder>
  - repaired/
  - QC/
    - fastqc/
    - multiqc/
  - pipeline_info/
```

Where:

<details markdown="1"><summary>Output folders</summary>

- `repaired` folder contains repaired fastq files and relative quality reports by [Wipertools](#wipertools)
- `QC` contains [FastQC](#fastqc) QC analysis results and [MultiQC](#multiqc) reports
- **[pipeline_info](#pipeline-information)** contains report metrics generated during the workflow execution

</details>

### Wipertools

[Wipertools](https://github.com/mazzalab/fastqwiper) is an ensemble method to recover corrupted FASTQ files, drop or fix pesky lines, remove unpaired reads, and settle reads interleaving.

<details markdown="1">
<summary>Output files</summary>

- `repaired/`
  - `mysampleA_R1_repaired.fastq.gz` and `mysampleA_R2_repaired.fastq.gz` are the repaired fastq files
  - `mysampleA_R1_report.txt` and `mysampleA_R2_report.txt` are the summaries of the cleaning task

</details>

### FastQC

[FastQC](http://www.bioinformatics.babraham.ac.uk/projects/fastqc/) gives general quality metrics about your sequenced reads. It provides information about the quality score distribution across your reads, per base sequence content (%A/T/G/C), adapter contamination and overrepresented sequences. For further reading and documentation see the [FastQC help pages](http://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/).

<details markdown="1">
<summary>Output files</summary>

- `fastqc/`
  - `*_fastqc.html`: FastQC report containing quality metrics.
  - `*_fastqc.zip`: Zip archive containing the FastQC report, tab-delimited data file and plot images.

</details>

### MultiQC

[MultiQC](http://multiqc.info) is a visualization tool that generates a single HTML report summarising all samples in your project. Most of the pipeline QC results are visualised in the report and further statistics are available in the report data directory.

Results generated by MultiQC collate pipeline QC from supported tools e.g. FastQC. The pipeline has special steps which also allow the software versions to be reported in the MultiQC output for future traceability. For more information about how to use MultiQC reports, see <http://multiqc.info>.

<details markdown="1">
<summary>Output files</summary>

- `multiqc/`
  - `multiqc_report.html`: a standalone HTML file that can be viewed in your web browser.
  - `multiqc_data/`: directory containing parsed statistics from the different tools used in the pipeline.
  - `multiqc_plots/`: directory containing static images from the report in various formats.

</details>

### Pipeline information

[Nextflow](https://www.nextflow.io/docs/latest/tracing.html) provides excellent functionality for generating various reports relevant to the running and execution of the pipeline. This will allow you to troubleshoot errors with the running of the pipeline, and also provide you with other information such as launch commands, run times and resource usage.

<details markdown="1">
<summary>Output files</summary>

- `pipeline_info/`
  - Reports generated by Nextflow: `execution_report.html`, `execution_timeline.html`, `execution_trace.txt` and `pipeline_dag.html`.
  - Reports generated by the pipeline: `pipeline_report.html`, `pipeline_report.txt` and `software_versions.yml`. The `pipeline_report*` files will only be present if the `--email` / `--email_on_fail` parameter's are used when running the pipeline.
  - Reformatted samplesheet files used as input to the pipeline: `samplesheet.valid.csv`.
  - Parameters used by the pipeline run: `params.json`.

</details>
